packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -394,6 +394,13 @@ class NavigatorObserver {
 /// manages a stack of [Route] objects and provides methods for managing
 /// the stack, like [Navigator.push] and [Navigator.pop].
 ///
+/// When your user interface fits this paradigm of a stack, where the user
+/// should be able to _navigate_ back to an earlier element in the stack,
+/// the use of routes and the Navigator is appropriate. On certain platforms,
+/// such as Android, the system UI will provide a back button (outside the
+/// bounds of your application) that will allow the user to navigate back
+/// to earlier routes in your application's stack.
```

Comment:
This leaves me with the question what I as a dev should do on non-android platforms.

Maybe a link to the appbar would be helpful that for most use cases shows a back button, if the navigator stack has content? See https://master-api.flutter.dev/flutter/material/AppBar/leading.html

Reply:
Good point - done.